generation/emitter/remap.rsp

```diff
@@ -244,6 +244,16 @@ FindFirstVolumeMountPointW::return=FindVolumeMointPointHandle
 FindNextVolumeMountPointA::hFindVolumeMountPoint=FindVolumeMointPointHandle
 FindNextVolumeMountPointW::hFindVolumeMountPoint=FindVolumeMointPointHandle
 FindVolumeMountPointClose::hFindVolumeMountPoint=FindVolumeMointPointHandle
+MapViewOfFile::return=MemoryMappedViewHandle
+MapViewOfFile3::return=MemoryMappedViewHandle
+MapViewOfFile3FromApp::return=MemoryMappedViewHandle
+MapViewOfFileEx::return=MemoryMappedViewHandle
+MapViewOfFileExNuma::return=MemoryMappedViewHandle
+MapViewOfFileFromApp::return=MemoryMappedViewHandle
+MapViewOfFileNuma2::return=MemoryMappedViewHandle
+UnmapViewOfFile::lpBaseAddress=MemoryMappedViewHandle
+UnmapViewOfFile2::BaseAddress=MemoryMappedViewHandle
+UnmapViewOfFileEx::BaseAddress=MemoryMappedViewHandle
```

Comment:
I wonder what the complications are from having two extra methods that release the handle. The RAIIRelease attribute will only point at the first one. So if the second and third would only be called manually, where would the caller have obtained the handle? I wonder if we should leave them accepting `IntPtr` instead of the new handle struct if that's the type that callers are likely to have.
I've never used any of these APIs, so it would be great to get the opinion of someone familiar with the area.

Reply:
You're supposed to use the UnmapViewOfFile that matches the MapViewOfFile API you called.

This one is interesting which is why I submitted a PR for review instead of just submitting this.

Reply:
Then I suppose we may need 3 handle types. And as the 2 and Ex methods take more than one parameter, they'll require special recognition by the projection so they know what to pass as the other argument(s).

Reply:
Actually looks like the docs say you can mix and match a bit: https://docs.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-unmapviewoffile#parameters. If you need the extra flags depending on the scenario, you can unmap with -Ex.

What does .NET do here? Do they just call the base unmap function regardless of how the handle was acquired?

Reply:
.NET only exposes one SafeHandle AFAIK, and it uses the simply-named release function. But that's to say nothing of which method was used to obtain the handle in the first place, as .NET doesn't expose it.